app/models/mixins/owner_mixin.rb

```diff
@@ -0,0 +1,13 @@
+module OwnerMixin
+  extend ActiveSupport::Concern
+
+  included do
+    validates :owner, :presence => true, :on => :create
+
+    before_validation :set_owner
```

Comment:
@lindgrenj6 doing some research on this.  Replacing `before_validation :set_owner` with `before_create` and adding a `Proc` to the conditional gets us around the if statement in the method.  
```
before_create :set_owner, :if => Proc.new { ManageIQ::API::Common::Request.current.present? }
```
The `Proc.new` only runs when the callback is called.  
Running locally with my tests it fixed the issue.

Reply:
Awesome, this looks much better than having the logic in the callback itself. Updated!